app/assets/javascripts/reducers/campaigns.js

```diff
@@ -26,7 +26,7 @@ export default function campaigns(state = initialState, action) {
     case RECEIVE_ALL_CAMPAIGNS: {
       const newState = {
         ...state,
-        all_campaigns: action.data.values
+        all_campaigns: action.data.campaigns
```

Comment:
This change is almost certainly what is causing the tests to fail.

Reply:
Yeah, right. But, I thought by just correcting this in the `all_campaigns` variable in the `campaign_editable.spec.jsx` would work. But it says that `all_campaigns` is undefined. So, I am still trying to figure out a way to resolve this.

Reply:
If you're looking for ways to debug this and understand what is happening with Redux state, I suggest (a) adding lots of console logging to see what the data looks like at each place in the flow, and (b) installing ReduxDevTools if you haven't already, to be able to inspect the Redux state at any time and see all the redux actions that transformed it.

Reply:
Have resolved this one. Will push the code soon.

Reply:
Basically, here the issue was with the `all_campaigns` state in `campaign_expandable.spec.jsx`. `allCampaigns` was being passed as a prop in `campaignEditable` component. But that was not even required as it is not a defined prop for it. I just correctly set the `all_campaigns` array in the initial state and the error was gone.